common/helpers.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Node, Edge } from 'reactflow';
+import { IComponent } from '../public/component_types';
+
+/**
+ * TODO: remove hardcoded nodes/edges.
+ *
+ * Converts the stored IComponents into the low-level ReactFlow nodes and edges.
+ * This may change entirely, depending on how/where the ReactFlow JSON will be
+ * persisted. Using this stub helper fn in the meantime.
+ */
+export function convertToReactFlowData(components: IComponent[]) {
+  const dummyNodes = [
+    {
+      id: 'semantic-search',
+      position: { x: 40, y: 10 },
+      data: { label: 'Semantic Search' },
+      type: 'group',
+      style: {
+        height: 110,
+        width: 700,
+      },
+    },
+    {
+      id: 'model',
+      position: { x: 25, y: 25 },
+      data: { label: 'Deployed Model ID' },
+      type: 'default',
+      parentNode: 'semantic-search',
+      extent: 'parent',
+    },
+    {
+      id: 'ingest-pipeline',
+      position: { x: 262, y: 25 },
+      data: { label: 'Ingest Pipeline Name' },
+      type: 'default',
+      parentNode: 'semantic-search',
+      extent: 'parent',
+    },
+  ] as Array<
+    Node<
+      {
+        label: string;
+      },
+      string | undefined
+    >
+  >;
+
+  const dummyEdges = [
+    {
+      id: 'e1-2',
+      source: 'model',
+      target: 'ingest-pipeline',
+      style: {
```

Comment:
I know this is going to change, but just to move conversation from offline to the github PR here. Want to confirm that we might potentially either have two different places or one from where this data will come form. One is through a redux call to an index where we have saved workflows with corresponding UI metadata. The other could be if we decide to go that way a library with fixed templates to pre-populate the node.

Reply:
Right, yes. The styling of different component types and edges will be standardized as we iterate and see what sort of base data we need for all component and edge types, and what separate fields we need to differ between them.